### PR TITLE
set release on native crash sdk 2.1.1

### PR DESCRIPTION
**Pull request by thinkocapo**
- v2.1.1 SDK now allows Release to have a version number
- added a section about two different ways to specify the Release
- added a 'Technical Notes' section for some gotcha's that Manoel explained, regarding Crash counts

### CODE REVIEW

File: app/src/main/java/com/example/vu/android/MyApplication.java [hunk 1]
```diff
@@ -20,14 +20,17 @@ public void onCreate() {
...
+
+            // we now enable this in AndroidManifest.xml
+            // options.setEnableSessionTracking(true);
+
             options.setBeforeSend((event, hint) -> {
 
                 //Remove PII
                 List<SentryException> exceptions = event.getExceptions();
-                if(exceptions.size() > 0){
+                if(exceptions != null && exceptions.size() > 0){
```
Comment (thinkocapo) [inline app/src/main/java/com/example/vu/android/MyApplication.java:31]: Manoel explained why we needed this.

File: app/src/main/java/com/example/vu/android/MyApplication.java [hunk 2]
```diff
@@ -20,14 +20,17 @@ public void onCreate() {
...
                     SentryException exception = exceptions.get(0);
-                    if(exception.getType().contains("NegativeArraySizeException")){
+                    if("NegativeArraySizeException".equals(exception.getType())) {
```
Comment (thinkocapo) [inline app/src/main/java/com/example/vu/android/MyApplication.java:33]: and this

File: app/build.gradle [hunk 3]
```diff
@@ -51,7 +51,7 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'androidx.appcompat:appcompat:1.1.0'
     implementation group: 'androidx.constraintlayout', name: 'constraintlayout', version: '1.1.3'
-    implementation 'io.sentry:sentry-android:2.1.0-beta.1'
+    implementation 'io.sentry:sentry-android:2.1.1'
```
Comment (thinkocapo) [inline app/build.gradle:54]: this is so Release number is captured when a Native Crash happens. available on 2.1.1

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -28,11 +28,23 @@
...
+        <!-- Default is 5seconds but we do 2 seconds for testing and demo's -->
+        <meta-data android:name="io.sentry.anr.timeout-interval-mills" android:value="3000" />
+
+        <!-- Easy way to get a new release if you're testing, to separate from past releases with lots of crashes/sessions. overrides what's in app/build.gradle -->
+        <!-- <meta-data android:name="io.sentry.release" android:value="io.sentry.sample@1.0.0+1" />-->
+
+        <!-- default interval for testing a session is 30seconds. Session starts when they open app. Session ends when user leaves app and it's idle for 30 seconds -->
+        <meta-data android:name="io.sentry.session-tracking.timeout-interval-millis" android:value="3000" />
+
+        <!-- Enables automatic session tracking. Disabled by default because back-end is not GA yet. SDK is GA. works for us for testing right now -->
```
Comment (thinkocapo) [inline app/src/main/AndroidManifest.xml:44]: this is new as of this month, roughly. Manoel pointed out all the things in here for me. Leaving them commented in here as they're not super obvious without the domain knowledge.

File: sentry.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
 # how a sentry.properties looks like
 defaults.org=testorg-az
-defaults.project=paranoid-android
+defaults.project=android
```
Comment (idosun) [inline sentry.properties:3]: @thinkocapo  I forget why we need a `sentry.properties` file for Android... ? Is this for the Gradle plugin?
In any case, **Makefile** and **sentry.properties** currently point to different projects (paranoid-android, android) - so align those if

Comment (thinkocapo) [reply sentry.properties]: @idosun thanks I made the update.

**Pull request merged**